File: iceoryx2-ffi/python/tests/file_name_tests.py
# Copyright (c) 2025 Contributors to the Eclipse Foundation
#
# See the NOTICE file(s) distributed with this work for additional
# information regarding copyright ownership.
#
# This program and the accompanying materials are made available under the
# terms of the Apache Software License 2.0 which is available at
# https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
# which is available at https://opensource.org/licenses/MIT.
#
# SPDX-License-Identifier: Apache-2.0 OR MIT

import iceoryx2_ffi_python as iceoryx2
import pytest


def test_file_name_can_be_constructed() -> None:
    sut = iceoryx2.FileName.new("some.file")
    assert sut.to_string() == "some.file"

def test_file_with_invalid_content_cannot_be_constructed() -> None:
    invalid_content = "/i/am/not/a/path"
    with pytest.raises(iceoryx2.SemanticStringError):
        sut = iceoryx2.FileName.new(invalid_content)

File: iceoryx2-bb/system-types/tests-nostd/src/file_name_tests.rs
// Copyright (c) 2023 Contributors to the Eclipse Foundation
//
// See the NOTICE file(s) distributed with this work for additional
// information regarding copyright ownership.
//
// This program and the accompanying materials are made available under the
// terms of the Apache Software License 2.0 which is available at
// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
// which is available at https://opensource.org/licenses/MIT.
//
// SPDX-License-Identifier: Apache-2.0 OR MIT

#![allow(clippy::disallowed_types)]

extern crate iceoryx2_bb_loggers;

use iceoryx2_bb_system_types_tests_common::file_name_tests;
use iceoryx2_bb_testing_nostd_macros::inventory_test;

#[inventory_test]
fn file_name_new_with_illegal_name_fails() {
    file_name_tests::file_name_new_with_illegal_name_fails();
}

#[inventory_test]
fn file_name_new_name_with_slash_is_illegal() {
    file_name_tests::file_name_new_name_with_slash_is_illegal();
}

#[inventory_test]
fn file_name_pop_fails_when_it_results_in_illegal_name() {
    file_name_tests::file_name_pop_fails_when_it_results_in_illegal_name();
}

#[inventory_test]
fn file_name_remove_fails_when_it_results_in_illegal_name() {
    file_name_tests::file_name_remove_fails_when_it_results_in_illegal_name();
}

#[inventory_test]
fn file_name_remove_range_fails_when_it_results_in_illegal_name() {
    file_name_tests::file_name_remove_range_fails_when_it_results_in_illegal_name();
}

#[inventory_test]
fn file_name_retain_fails_when_it_results_in_illegal_name() {
    file_name_tests::file_name_retain_fails_when_it_results_in_illegal_name();
}
